airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py

```diff
@@ -223,9 +269,12 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
         Thus all the slices data are compared to the initial state, but only on the last one we write it to the stream state.
         This approach gives us the maximum state value of all the records and we exclude the state updates between slice processing
         """
-        if not current_stream_state:
-            current_stream_state = {self.cursor_field: self.start_date}
-        return {self.cursor_field: max(latest_record.get(self.cursor_field, ""), current_stream_state.get(self.cursor_field, ""))}
+        if self.first_run:
```

Comment:
we can check if `self.initial_state is None` instead and remove the `first_run` field